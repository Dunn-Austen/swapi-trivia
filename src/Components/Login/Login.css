.login-section {
  align-items: center;
  display: flex;
  justify-content: center;
  height: 100%;
  width: 100%;
}

.login-home {
  background-color: #8B837B;
  border: 1px solid #5ae273;
  border-radius: 400px;
  height: 300px;
  width: 360px;
}

.form-container {
  display: flex;
  height: 350px;
  justify-content: center;
  flex-direction: column;
  text-align: center;
  width: 100%;
}

h1 {
  font-size: 20px;
}

h2 {
  color: #5ae273;
  font-size: 20px;
}

form {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 230px;
  justify-content: space-around;
}

input {
  border: .1px solid #5ae273;
  font-size: 20px;
  height: 20px;
  text-indent: 2px;
  width: 150px;
}

textarea {
  border: .1px solid #5ae273;
  font-size: 20px;
  height: 40px;
  text-indent: 2px;
  width: 150px;
}

label {
  display: block;
  font-size: 20px;
  margin-bottom: 5px;
}

select {
  border: 1px solid #5ae273;
  font-size: 15px;
  height: 25px;
  width: 120px;
}

button {
  height: 40px;
}

button:hover {
  border: .1px solid #5ae273;
}
