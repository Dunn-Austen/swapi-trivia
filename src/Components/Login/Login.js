import React, { Component } from 'react'
import {Link, withRouter} from 'react-router-dom'
import './Login.css'


export class Login extends React.Component {
    constructor() {
        super();
        this.state = {
                name: '',
                quote: '',
                ranking: '',
                isSignedIn:false,
                errors: ''
        }
    }

    handleChange(event) {
        this.setState({ [event.target.name]: event.target.value })
    }

  login = (event) => {
        event.preventDefault();
        const {name,quote,ranking}=this.state
        if (!name || name.length === 0 || !quote || quote.length === 0) {
            return this.setState({ error: 'Please fill out all inputs!!' });
        }
    console.log(this.props.history.push("./movies"))
      this.setState({isSignedIn:true})
      this.props.enterUserInfo({ name, quote, ranking })

    }

    render() {
        return (
          <section className="login-section">
            <div className ='login-home'>
              <div className="form-container">
                <h1 className ='welcome'>Welcome to Star Wars!!!</h1>
                <h2>{this.state.error}</h2>
                <form>
                  <input type='text' placeholder='Name' value={this.state.name} name = 'name' onChange={event => this.handleChange(event)} />
                  <textarea type='text' placeholder='Quote' value={this.state.quote} name = 'quote' onChange={event => this.handleChange(event)}></textarea>
                  <div>
                    <label>Select Level:</label>
                      <select value={this.state.ranking} name = 'ranking' onChange={event => this.handleChange(event)}>
                        <option>Padawan</option>
                        <option>Jedi knight</option>
                        <option>Jedi master</option>
                      </select>
                  </div>
                  <button type='submit' className='loginButton' onClick={this.login}>Luke I Am Your Father</button>
                </form>
              </div>
            </div>
          </section>
        )
    }
}

export default withRouter(Login)

Login.propTypes = {
  enterUserInfo: PropTypes.func
}
