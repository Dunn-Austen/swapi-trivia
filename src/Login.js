import React, {Component} from 'react'
import './Login.css'


class Login extends Component {
    constructor() {
        super();
        this.state = {
            user: {
                name: '',
                quote: '',
                ranking: ''
              }
        }
    }

    render() {
        return (
            <div className = 'loginHome'>
             <h1 className = 'welcome'>Welcome to Star Wars!!!</h1>
              <form>
                    <input />
                    <div>
                        <label>Select Level</label>
                        <select></select>
                    </div>
                    <button></button>
              </form>
            </div>
        )
    }
}

export default Login
