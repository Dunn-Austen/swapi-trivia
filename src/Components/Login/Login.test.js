import React from 'react';
import { shallow } from 'enzyme'
import { Login } from './Login';

describe('Login', () => {
  let wrapper;

    beforeEach(() => {
    wrapper = shallow(<Login enterUserInfo={jest.fn()}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is called', () => {
    const mockEvent = { target: { name: 'name', value: 'foster'} };
    const expected = 'foster';
    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state('name')).toEqual(expected);
    //Tried relocating let wrapper = shallow(<Login />) locally to each test to no avail
    //Wrapper continues to display as null (can't read handleChange of 'null')
    //Tried chaining the .dive() method to the shallow instance without luck, wrapper is still null
    //Tried to intialize then expectba new variable (let wrapper2 = shallow(<Login />)), which didn't work (still null)
    //Tried tweaking the shallow import syntax
    //Tried removing the initial snapshot test in case it was causing any contamination
    //SOLVED - sort of, I altered the Login import from (import Login to { Login} , but now the snapshot fails)
  });

  it('should run handleChange when the inputs are manipulated', () => {
    const mockEvent = { target: { name: 'name', value: 'foster'} };
    wrapper.instance().handleChange = jest.fn();
    // const mockEvent = { preventDefault: jest.fn() };

    wrapper.find('input').simulate('change', mockEvent);

    expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent);
  });

  it('should invoke login upon button click', () => {
    const mockEvent = { preventDefault: jest.fn() };
    wrapper.instance().login = jest.fn();
    wrapper.instance().forceUpdate();

    wrapper.find('button').simulate('click', mockEvent);

    expect(wrapper.instance().login).toHaveBeenCalledWith(mockEvent);
  });

  it('should throw error if login if validations are not satisfied', () => {
    const mockEvent = { preventDefault: jest.fn() };
    const originalState = {isSignedIn: false}
    const expected = {error: "Please fill out all inputs!!", errors: "", isSignedIn: false, name: "", quote: "", ranking: "Padawan"};
    // wrapper.instance().login = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.instance().setState(originalState);

    wrapper.instance().login(mockEvent);

    expect(wrapper.state()).toEqual(expected);
  });

  it('should update state when login is called', () => {
    const mockEvent = { preventDefault: jest.fn() };
    const originalState = {name: 'Tom', quote: 'Nah', ranking: 'Padawan', isSignedIn: false, errors: ''};
    const expected = {name: 'Tom', quote: 'Nah', ranking: 'Padawan', isSignedIn: true, errors: ''};
    // wrapper.instance().login = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.instance().setState(originalState);

    wrapper.instance().login(mockEvent);

    expect(wrapper.state()).toEqual(expected);
  });

});
