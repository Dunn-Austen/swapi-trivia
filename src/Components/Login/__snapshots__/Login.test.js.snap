// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
<section
  className="login-section"
>
  <div
    className="login-home"
  >
    <div
      className="form-container"
    >
      <h1
        className="welcome"
      >
        Welcome to Star Wars!!!
      </h1>
      <h2 />
      <form>
        <input
          name="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value=""
        />
        <textarea
          name="quote"
          onChange={[Function]}
          placeholder="Quote"
          type="text"
          value=""
        />
        <div>
          <label>
            Select Level:
          </label>
          <select
            name="ranking"
            onChange={[Function]}
            value="Padawan"
          >
            <option>
              Padawan
            </option>
            <option>
              Jedi knight
            </option>
            <option>
              Jedi master
            </option>
          </select>
        </div>
        <button
          className="loginButton"
          onClick={[Function]}
          type="submit"
        >
          Luke I Am Your Father
        </button>
      </form>
    </div>
  </div>
</section>
`;
